### PR TITLE
Fixed the abort_job random EOFError.

### PR DESCRIPTION
**Pull request by yhwen**

### CODE REVIEW

File: nvflare/private/fed/client/command_agent.py [hunk 1]
```diff
@@ -46,34 +48,43 @@ def start(self, fl_ctx: FLContext):
...
+                self.logger.info("listener communication terminated.")
+                break
             except Exception as e:
                 # traceback.print_exc()
-                print(f"Process communication exception: {self.listen_port}.")
-            finally:
-                conn.close()
+                self.logger.error(f"Process communication error: {self.listen_port}: {e}.", exc_info=False)
 
+        try:
```
Comment (chesterxgchen) [inline nvflare/private/fed/client/command_agent.py:81]: should this be in finally part ?

File: nvflare/private/fed/server/server_command_agent.py [hunk 2]
```diff
@@ -43,34 +43,40 @@ def start(self, engine):
 
     def listen_command(self, engine):
         try:
```
Comment (chesterxgchen) [inline nvflare/private/fed/server/server_command_agent.py:45]: this code seems to be duplicate as above, consider wrap into a function ?

File: nvflare/private/fed/server/server_command_agent.py [hunk 3]
```diff
@@ -43,34 +43,40 @@ def start(self, engine):
 
     def listen_command(self, engine):
```
Comment (chesterxgchen) [inline nvflare/private/fed/server/server_command_agent.py:44]: sever_command_agent.py  and command_agent.py in large part are duplicate, consider refactoring this ?

Comment (yhwen) [reply nvflare/private/fed/server/server_command_agent.py:44]: The processing logic are similar, but not duplicate. I will try to extract the common part.

File: nvflare/private/fed/server/server_command_agent.py (hunk 2 above)
Comment (yhwen) [reply nvflare/private/fed/server/server_command_agent.py]: done

File: nvflare/private/fed/server/server_command_agent.py (hunk 3 above)
Comment (yhwen) [reply nvflare/private/fed/server/server_command_agent.py]: done

File: nvflare/private/fed/client/command_agent.py [hunk 4]
```diff
@@ -36,51 +38,60 @@ def __init__(self, federated_client, listen_port, client_runner) -> None:
...
-        self.thread = threading.Thread(target=listen_command, args=[self, fl_ctx])
+        engine = fl_ctx.get_engine()
+        self.thread = threading.Thread(
+            target=listen_command, args=[self.listen_port, engine, self.execute_command, self.logger]
+        )
         self.thread.start()
 
-        pass
-
     def listen_command(self, fl_ctx):
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/client/command_agent.py:50]: this `listen_command` is not being used

Comment (YuanTingHsieh) [reply nvflare/private/fed/client/command_agent.py:50]: we use the one from `from ..utils.fed_utils import listen_command`

**Review by YuanTingHsieh — CHANGES_REQUESTED** [review]
As commented

File: nvflare/private/fed/client/command_agent.py (hunk 4 above)
Comment (yhwen) [reply nvflare/private/fed/client/command_agent.py]: Removed.

Comment (yhwen) [reply nvflare/private/fed/client/command_agent.py]: Removed.

**Review by YuanTingHsieh — APPROVED** [review]

**Pull request merged**